src/js/other-websites/inpage_menu.js

```diff
@@ -51,28 +51,386 @@ async function isUserSignedIn() {
+  );
+
+  maskResults.forEach((maskResult) => {
+    const emailAddress = maskResult.dataset.mask;
+    const label = maskResult.dataset.label;
+    // const labelElement = maskResult.querySelectorAll(".relay-email-address-label")[0];
+    // const label = labelElement ? labelElement.dataset.label : "";
+    const matchesSearchFilter =
+      label.toLowerCase().includes(query.toLowerCase()) ||
+      emailAddress.toLowerCase().includes(query.toLowerCase());
+
+    if (matchesSearchFilter) {
+      maskResult.classList.remove("is-hidden");
+    } else {
+      maskResult.classList.add("is-hidden");
+    }
+  });
+
+  // TODO: Add #/# label
+  // filterLabelVisibleCases.textContent = aliasContainers.filter(aliasContainer => !aliasContainer.classList.contains("is-hidden")).length;
+  // filterLabelTotalCases.textContent = aliasContainers.length;
+}
+
+const buildContent = {
+  loggedIn: {
+    free: async (relaySiteOrigin) => {
+      const fxRelayMenuBody = document.getElementById("fxRelayMenuBody");
+
+      const signedInContentFree = document.querySelector(
+        ".fx-content-signed-in-free"
+      );
+
+      const signedInContentPremium = document.querySelector(
+        ".fx-content-signed-in-premium"
+      );
+
+      // Create "You have .../.. remaining relay address" message
+      const remainingAliasesSpan = document.querySelector(
+        ".fx-relay-menu-remaining-aliases"
+      );
+
+      const { maxNumAliases } = await browser.storage.local.get(
+        "maxNumAliases"
+      );
+
+      const maskLists = document.querySelectorAll(".fx-relay-menu-masks-list");
+      const masks = await getMasks();
+
+      maskLists?.forEach(async (maskList) => {
+        // Set Mask List label names
+        const label = maskList.querySelector(".fx-relay-menu-masks-list-label");
+        const stringId = label.dataset.stringId;
+        label.textContent = browser.i18n.getMessage(stringId);
+
+        // Populate mask lists
+        await populateMaskList(maskList, masks);
+      });
+
+      const numAliasesRemaining = maxNumAliases - masks.length;
+      const maxNumAliasesReached = numAliasesRemaining <= 0;
+
+      // Create "Get unlimited aliases" button
+      const getUnlimitedAliasesBtn = document.querySelector(
+        ".fx-relay-menu-get-unlimited-aliases"
+      );
+
+      const generateAliasBtn = document.querySelector(
+        ".fx-relay-menu-generate-alias-btn"
+      );
+
+      // Free user: Set text informing them how many aliases they can create
+      remainingAliasesSpan.textContent = browser.i18n.getMessage(
+        "popupRemainingAliases_2_mask",
+        [numAliasesRemaining, maxNumAliases]
+      );
+
+      getUnlimitedAliasesBtn.textContent = browser.i18n.getMessage(
+        "popupGetUnlimitedAliases_mask"
+      );
+
+      // Create "Get unlimited aliases" link
+      getUnlimitedAliasesBtn.href = `${relaySiteOrigin}/premium?utm_source=fx-relay-addon&utm_medium=input-menu&utm_content=get-premium-link`;
```

Comment:
Thought: might it be easier to reason about this if we used the same strategy as for the different types of users here, i.e. just add the different buttons to the HTML and displaying the relevant one, rather than setting and re-setting the different attributes here depending on the circumstances, and hoping that we don't forget to clear some attribute or another?

Reply:
Migrated to helper functions here: https://github.com/mozilla/fx-private-relay-add-on/blob/mpp-830-sign-back-in-page/src/js/other-websites/inpage_menu.js#L517